packages/patternfly-4/react-core/src/demos/BulkSelectTable.md

```diff
@@ -0,0 +1,225 @@
+      const id = rows[rowId].id;
+      rows[rowId].selected = isSelected;
+      this.setState({
+        res: rows,
+        selectedItems: isSelected ? [...selectedItems, id] : selectedItems.filter(itemId => itemId !== id)
+      });
+    };
+
+    this.updateSelected = () => {
+      const { res, selectedItems } = this.state;
+      let rows = res.map(post => {
+        post.selected = selectedItems.includes(post.id);
+        return post;
+      });
+
+      this.setState({
+        res: rows
+      });
+    };
+
+    this.handleSelectClick = newState => {
+      if (newState === 'none') {
+        this.setState(
+          {
+            selectedItems: []
+          },
+          this.updateSelected
+        );
+      } else if (newState === 'page') {
+        let newRows = [];
+        let rows = this.state.res.map(post => {
+          const isSelected = post.selected;
+          newRows = isSelected ? [...newRows] : [...newRows, post.id];
+          post.selected = true;
+          return post;
+        });
+        this.setState(
+          {
+            selectedItems: this.state.selectedItems.concat(newRows)
+          },
+          this.updateSelected
+        );
+      } else {
+        const { total } = this.state;
+        let newRows = [];
+        for (var i = 1; i <= total; i++) newRows = [...newRows, i];
+
+        this.setState(
+          {
+            selectedItems: newRows
+          },
+          this.updateSelected
+        );
+      }
+    };
+
+    this.onDropDownToggle = isOpen => {
+      this.setState({
+        isDropDownOpen: isOpen
+      });
+    };
+
+    this.onDropDownSelect = event => {
+      this.setState({
+        isDropDownOpen: !this.state.isDropDownOpen
+      });
+    };
+  }
+
+  fetch(page, perPage) {
+    this.setState({ loading: true });
+    fetch(`https://jsonplaceholder.typicode.com/posts?_page=${page}&_limit=${perPage}`)
+      .then(resp => resp.json())
+      .then(resp => this.setState({ res: resp, perPage, page, loading: false }))
+      .then(() => this.updateSelected())
+      .catch(err => this.setState({ error: err, loading: false }));
+  }
+
+  componentDidMount() {
+    this.fetch(this.state.page, this.state.perPage);
+  }
+
+  renderPagination(variant = 'top') {
+    const { page, perPage, total } = this.state;
+    return (
+      <Pagination
+        itemCount={total}
+        page={page}
+        perPage={perPage}
+        onSetPage={(_evt, value) => {
+          this.fetch(value, perPage);
+        }}
+        onPerPageSelect={(_evt, value) => {
+          this.fetch(1, value);
+        }}
+        variant={variant}
+      />
+    );
+  }
+
+  buildSelectDropdown() {
+    const { isDropDownOpen, selectedItems, total } = this.state;
+    const numSelected = selectedItems.length;
+    const allSelected = numSelected === total;
+    const anySelected = numSelected > 0;
+    const isChecked = allSelected ? true : numSelected > 0 ? null : false;
```

Comment:
Can we avoid using nested ternary? Breaking the statements out into separate vars would help the readability, at the least we could include parenthesis around the expressions to make it easier to follow with the eyeballs.